packages/app/pages/EconomicEvent/EconomicEvent.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+
+import allEconomicEvents from "../../../ui-bindings/EconomicEvent/AllEconomicEvents";
+import economicEventById from "../../../ui-bindings/EconomicEvent/EconomicEventById";
+
+/**
+ * Main component for the page. Contains a search box and a list of
+ * all available Economic Events
+ */
+class EconomicEvent extends React.Component {
+
+  // declare the state and what variables are used
+  state = {
+    eventId: 57
+  };
+
+  // function to handle the search action on the search box
+  handleClick = (event) => {
+    // do not submit the form in the URL
+    event.preventDefault();
+
+    // fetch the economic event number from the input box
+    let eventId = document.getElementById("idForm").value;
+
+    // save the economic event id in the state
+    this.setState({eventId: eventId});
+  };
+
+  // Draw the component on the screen
+  render() {
+    return (
+      <div>
+        <h1>Economic Events</h1>
+
+        <h3>Search by ID:</h3>
+        <form onSubmit={this.handleClick}>
+          <input type="text" id="idForm" defaultValue="57"/>
+        </form>
+
+        <h4>Results</h4>
+        ======================================================<br/>
+        <EconomicEventById eventId={this.state.eventId} />
```

Comment:
While using elements before they appear is allowed by typescript, it is disallowed by our Linter. please move the declaration for this( and the other methods you use above the class to be in compliance with the linter

Reply:
I moved the individual components above the React.Component for the page. If this is not what you meant, please let me know

Reply:
Sorry, I was really vague. our Linter disallows using react elements (i.e. "EconomicEventById") before they are declared in the file.